pkg/actuators/awsplacementgroup/controller_suite_test.go

```diff
@@ -0,0 +1,58 @@
+package awsplacementgroup
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	machinev1 "github.com/openshift/api/machine/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+var (
+	cfg     *rest.Config
+	testEnv *envtest.Environment
+
+	ctx = context.Background()
+)
+
+func TestReconciler(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecsWithDefaultAndCustomReporters(t,
+		"Controller Suite",
+		[]Reporter{printer.NewlineReporter{}})
+}
+
+var _ = BeforeSuite(func() {
+	testEnv = &envtest.Environment{
+		CRDDirectoryPaths: []string{filepath.Join("..", "..", "..", "vendor", "github.com", "openshift", "api", "machine", "v1")},
+	}
+	machinev1.Install(scheme.Scheme)
+
+	var err error
+	cfg, err = testEnv.Start()
+	Expect(err).ToNot(HaveOccurred())
+	Expect(cfg).ToNot(BeNil())
+})
+
+var _ = AfterSuite(func() {
+	Expect(testEnv.Stop()).To(Succeed())
+})
+
+// StartTestManager adds recFn
+func StartTestManager(mgr manager.Manager) context.CancelFunc {
+	mgrCtx, cancel := context.WithCancel(ctx)
+	go func() {
+		defer GinkgoRecover()
+
+		Expect(mgr.Start(mgrCtx)).To(Succeed())
+	}()
+	return cancel
+}
```

Comment:
If you have an additional channel here, which you close after the mgr has stopped, you can ensure your tests wait for the manager to have stopped before they move on to the next